cmd/auth/main.go

```diff
@@ -240,40 +247,73 @@ func newService(db *sqlx.DB, tracer opentracing.Tracer, secret string, logger lo
 	return svc
 }
 
-func startHTTPServer(tracer opentracing.Tracer, svc auth.Service, port string, certFile string, keyFile string, logger logger.Logger, errs chan error) {
+func startHTTPServer(ctx context.Context, tracer opentracing.Tracer, svc auth.Service, port string, certFile string, keyFile string, logger logger.Logger) error {
 	p := fmt.Sprintf(":%s", port)
-	if certFile != "" || keyFile != "" {
+	server := &http.Server{Addr: p, Handler: httpapi.MakeHandler(svc, tracer)}
+	errCh := make(chan error)
+	proto := "http"
+	switch {
+	case certFile != "" || keyFile != "":
 		logger.Info(fmt.Sprintf("Authentication service started using https, cert %s key %s, exposed port %s", certFile, keyFile, port))
-		errs <- http.ListenAndServeTLS(p, certFile, keyFile, httpapi.MakeHandler(svc, tracer))
-		return
+		go func() {
+			errCh <- server.ListenAndServeTLS(certFile, keyFile)
+		}()
+		proto = "https"
+	default:
+		logger.Info(fmt.Sprintf("Authentication service started using http, exposed port %s", port))
+		go func() {
+			errCh <- server.ListenAndServe()
+		}()
 	}
-	logger.Info(fmt.Sprintf("Authentication service started using http, exposed port %s", port))
-	errs <- http.ListenAndServe(p, httpapi.MakeHandler(svc, tracer))
 
+	select {
+	case <-ctx.Done():
+		if err := server.Shutdown(context.Background()); err != nil {
+			logger.Error(fmt.Sprintf("Authentication %s service error occured during shutdown at %s: %s", proto, p, err))
+			return fmt.Errorf("Authentication %s service error occured during shutdown at %s: %w", proto, p, err)
+		}
+		logger.Info(fmt.Sprintf("Authentication %s service shutdown of http at %s", proto, p))
+		return nil
```

Comment:
Please use context with the deadline here to prevent infinite waiting for shutdown:
```suggestion
c, done := context.WithTimeout(ctx, time.Second)
defer done()
if err := server.Shutdown(c); err != nil {
	logger.Error(fmt.Sprintf("Authentication %s service error occured during shutdown at %s: %s", proto, p, err))
	return fmt.Errorf("Authentication %s service error occured during shutdown at %s: %w", proto, p, err)
}
logger.Info(fmt.Sprintf("authentication %s service shutdown of http at %s", proto, p))
return nil
```

Reply:
shall We have env variable to set timeout duration or let it be constant for 1 or x  second

Reply:
I think constant (~1s) is fine. We don't have to configure everything, since the user already has a lot of env vars to take care of. I also prefer constant over magic numbers.